docs/csharp/how-to/modify-string-contents.md

```diff
@@ -0,0 +1,76 @@
+
+Regular expressions are most useful for searching and replacing text that follows a pattern, rather than known text. See [How to: search strings](search-strings.md) for more details. The search pattern, "the\s" searches for the word "the" followed by a white-space character. That part of the pattern ensures that it doesn't match "there" in the source string. For more information on regular expression language elements, see [Regular Expression Language - Quick Reference](../../../standard/base-types/regular-expression-language-quick-reference.md).
+
+[!code-csharp-interactive[replace creates a new string](../../../samples/snippets/csharp/how-to/strings/ModifyStrings.cs#5)]  
+
+The <xref:System.Text.StringBuilder.ToString%2A?displayProperty=nameWithType> method returns an immutable string with the contents in the <xref:System.Text.StringBuilder> object.
+
+## Modifying individual characters
+
+You can produce a character array from a string, modify the contents of the array, and then create a new string from the modified contents of the array. 
+
+The following example shows how to replace a set of characters in a string. First, it uses the <xref:System.String.ToCharArray?displayProperty=nameWithName> method to create an array of characters. Then, it uses the <xref:System.String.IndexOf%2A> method to find the starting index of the word "fox." Then, the next three characters are replaced with a different word. Finally, a new string is constructed from the updated character array.
+
+[!code-csharp-interactive[replace creates a new string](../../../samples/snippets/csharp/how-to/strings/ModifyStrings.cs#6)]  
+
+## Unsafe modifications to string
+
+Using **unsafe** code, you can modify a string "in place" after it has been created. Unsafe code bypasses many of the features of .NET designed to minimize certain types of bugs in code. You need to use unsafe code to modify a string in place because the string class is designed as an **immutable** type. Once it has been created, its value does not change. Unsafe code circumvents this property by accessing and modifying the memory used by a `string` without using normal `string` methods.
+The following example is provided for those rare situations where you want to modify a string in-place using unsafe code. The example shows how to use the `fixed` keyword. The `fixed` keyword prevents the garbage collector (GC) from moving the string object in memory while code accesses the memory using the unsafe pointer. It also demonstrates one possible side effect of unsafe operations on strings that results from the way that the C# compiler stores (interns) strings internally. In general, you shouldn't use this technique unless it is absolutely necessary. You can learn more in the articles on [unsafe](../langauge-reference/keywords/unsafe.md) and [fixed](../language-reference/keywords/fixed.md).
```

Comment:
It might be useful to link to some information on interning strings: (\[interns\](\<xref:System.String.Intern%2A>))

Reply:
Good suggestion. I looked and could not find a good topi in our conceptual docs.